src/Adapter/MailTemplate/MailTemplateDataProvider.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\MailTemplate;
+
+use Language;
+
+/**
+ * This class will provide data from filesystem about MailTemplate.
+ */
+class MailTemplateDataProvider
+{
+    /**
+     * @var Language[]
+     */
+    private $languages;
+
+    public function __construct(array $languages)
+    {
+        $this->languages = $languages;
+    }
+
+    /**
+     * Get all templates.
+     *
+     * @return array Suppliers
+     */
+    public function getTemplates()
+    {
+        $mailsTemplates = [];
+
+        foreach ($this->languages as $language) {
+            $languageIsoCode = $language['iso_code'];
+            $languageId = $language['id_lang'];
+
+            $mailsPath = _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR . 'mails' . DIRECTORY_SEPARATOR . $languageIsoCode;
```

Comment:
Part of this variable can be generate once.

Reply:
:+1: